Allow operator to install resources to WATCH_NAMESPACE instead of requiring keda ns

joelsmith opened the pull request:
Currently the KEDA OLM operator is hard-coded to work only in the keda namespace.

This change makes it so that whichever namespace the KEDA operator is installed to,
the operands (and required resources) will be installed to the same namespace.

The fundamental difference:
Before:
* The operator could be installed in any namespace
* The kedacontroller custom resource could only be installed in the `keda` namespace.
* The operand controllers and necessary resources were always installed (by the operator) in the `keda` namespace.

After:
* The operator can be installed in any namespace
* The kedacontroller custom resource can only be installed in the same namespace where the operator is installed.
* The operand controllers and necessary resources are always installed (by the operator) in the namespace where the operator is installed.

### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)
- [x] Add tests for transform function
- [x] Update docs to explain the new behavior

zroubalik commented on controllers/keda/configmap_controller.go:
```diff
@@ -36,18 +36,19 @@ import (
-	Scheme *runtime.Scheme
+	Log                logr.Logger
+	Scheme             *runtime.Scheme
+	configMapNamespace string
```
might be better to give this a better name, the configmap there is a bit confusing

joelsmith replied:
will do

zroubalik reviewed (commented):
Looking good

joelsmith commented:
@zroubalik thanks for the review! I had a few final tasks which are now done. (I renamed a variable as suggested by your review comment, I switched the unit tests to use ginkgo to get rid of a CI test error, and I updated the docs to match the new behavior.)

If you have time for a final review, that would be great. If not, I can have somebody from my team review it -- now that I know you're okay with the general approach.

zroubalik reviewed (approved):
LGTM

Great job! I like the change in the unit tests
